test/controller/ATSPAYEDataControllerTest.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controller
+
+import controllers.ATSPAYEDataController
+import org.mockito.Matchers.{eq => eqTo, _}
+import org.mockito.Mockito._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.time.{Millis, Seconds, Span}
+import play.api.test.FakeRequest
+import services.NpsService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+import utils.TestConstants._
+
+import scala.concurrent.Future
+
+class ATSPAYEDataControllerTest extends UnitSpec with MockitoSugar with WithFakeApplication with ScalaFutures {
+
+  implicit val defaultPatience =
+    PatienceConfig(timeout = Span(5, Seconds), interval = Span(500, Millis))
+
+  class TestController extends ATSPAYEDataController {
+    val request = FakeRequest()
+    override lazy val npsService: NpsService = mock[NpsService]
+  }
+
+  "getAtsData" should {
+
+    "return a failed future" in new TestController {
```

Comment:
MIssing a happy path test

Reply:
Done... What is the best way to create any old json value for testing in play?

Reply:
The code is very basic at the minute, I would have my connector serialize any json to a scala object so then you would just create scala objects in the normal way and return them from any mocks. Then the only place that needs json is the connector and can either deserialize a scala object or have some json as a string or in a file that you load in

Reply:
The code is basic on purpose. it is a tracer bullet to put an architectural skeleton in place in order to support the PAYE journey... It will be fleshed out as we play subsequent stories.